### PR TITLE
[DROOLS-3496][DROOLS-3473][DROOLS-3487] Fix external JUnit runner support for DMN scenarios

### PR DESCRIPTION
**Pull request by danielezonca**
This PR add DMN-related dependencies to user project and improve error message in case of broken KieContainer.

https://issues.jboss.org/browse/DROOLS-3496
https://issues.jboss.org/browse/DROOLS-3473
https://issues.jboss.org/browse/DROOLS-3487

NOTE 1: remove ScenarioJunitActivatorTest in an existing project and create/save a scenario to force pom dependency refresh
NOTE 2: KieContainer is broken if the DMN file has a compilation error or if there are more than a single DMN file with same `namespace` and `name`. Use `Build` button in the asset list page to force KieContainer refresh when the error has been fixed
NOTE 3: to build user project from outside using `mvn package` command, you need first to install locally `drools-wb-scenario-simulation-editor`. If you have a compilation error related to `kie-maven-plugin` version please change it to a final version like `7.17.0.Final` because it seems like we have no SNAPSHOT version for that component on nexus 

@kkufova @gitgabrio @Rikkola

### CODE REVIEW

**Comment by danielezonca** [comment]
jenkins retest this please

**Comment by danielezonca** [comment]
jenkins retest this please

**Comment by danielezonca** [comment]
jenkins retest this please

**Comment by danielezonca** [comment]
jenkins retest this please

File: drools-wb-screens/drools-wb-scenario-simulation-editor/drools-wb-scenario-simulation-editor-backend/src/main/java/org/drools/workbench/screens/scenariosimulation/backend/server/AbstractKieContainerService.java [hunk 1]
```diff
@@ -33,6 +33,11 @@
 
     protected KieContainer getKieContainer(Path path) {
         KieModule kieModule = moduleService.resolveModule(path);
-        return buildInfoService.getBuildInfo(kieModule).getKieContainer();
+        KieContainer kieContainer = buildInfoService.getBuildInfo(kieModule).getKieContainer();
+        if (kieContainer == null) {
+            throw new IllegalArgumentException("Impossible to retrieve KieContainer, please fix compilation errors of " +
```
Comment (kkufova) [inline drools-wb-screens/drools-wb-scenario-simulation-editor/drools-wb-scenario-simulation-editor-backend/src/main/java/org/drools/workbench/screens/scenariosimulation/backend/server/AbstractKieContainerService.java]: Change message to:

> Retrieving KieContainer has failed. Fix all compilation errors within the project and build the project again.

**Comment by danielezonca** [comment]
jenkins retest this please

**Comment by danielezonca** [comment]
jenkins retest this please

**Comment by danielezonca** [comment]
jenkins retest this please

**Comment by danielezonca** [comment]
@manstis 

Can you please squash and merge?

**Pull request merged**